Inline Help: Add Managing Purchases Link

Aurorum opened the pull request:
#### Changes proposed in this Pull Request

Includes the link to [this document](https://wordpress.com/support/manage-purchases/) on the requested sections. The only exception is for the My Home: @michaeldcain - Inline Help appears to be hidden for the My Home section, should it be made visible?

As requested, it's only visible for people who have made a purchase, unless they're on the Purchases section (where the link is already present) as I think that's what the original issue implies.

#### Testing instructions

Check the sections mentioned in the original issue, and verify the "Managing Purchases, Renewals, and Cancellations" link appears if you have a purchase which you can manage/renew/cancel.

<img width="501" alt="Screenshot 2020-09-04 at 17 09 15" src="https://user-images.githubusercontent.com/43215253/92261501-6ea36900-eed1-11ea-9d77-f97b99adfb03.png">

cc @michaeldcain 

Fixes #45383

Aurorum committed:
Add "Managing Purchases, Renewals, and Cancellations" links

Aurorum committed:
Logic for when to display Managing Purchases link

fditrapani commented:
Thanks for taking this on @Aurorum! 

 I checked the screens for sites on the Business, and Premium, plan and I'm not seeing the links. The plans I own don't have payment methods associated to them since they were bought with credits so I wonder if that's why? It could be my setup so let's see if anyone else runs into any issues.

Aurorum commented:
Hi! I suspect that the set-up with credits might be the reasoning behind that, but let me know if that's still an issue on an account with no payment methods. :)

sirbrillig commented:
This is great! I tested the following routes, which I deduced from the description in the original issue. Hopefully I got them right.

I see the link appear in the question mark menu on these pages.

- /plans/:siteId
- /plans/my-plan/:siteId
- /settings/:siteId

I do not see the link on these pages.

- /help (I don't see the question mark icon at all on this page)
- /home/:siteId (I don't see the question mark icon at all on this page, but I see that you already asked @michaeldcain about this)
- /me/purchases/:siteId/:receiptId

Is there anything you think I might be doing wrong?

michaeldcain commented:
> Inline Help appears to be hidden for the My Home section, should it be made visible?

Sorry I missed this question originally. I believe this is intentional, and instead, the Customer Home page has a card for help topics:

<img width="398" alt="Screen Shot 2020-09-17 at 3 44 48 PM" src="https://user-images.githubusercontent.com/942359/93520680-db205e00-f8fc-11ea-96ca-d7d7631f9037.png">

Would it be possible to add it there?

Aurorum committed:
Add link to My Home

Aurorum committed:
Add link to Domains section as well

Aurorum commented:
Thank you for the clarification, that makes sense! I've added it to My Home, and I've also added that link to the Domains section, as mentioned by @fditrapani in the original issue. :)  

<img width="395" alt="Screenshot 2020-09-19 at 13 37 20" src="https://user-images.githubusercontent.com/43215253/93667362-47c26680-fa7d-11ea-8980-2ff7568c5fd8.png">

sirbrillig commented:
Tested again, following the updated list from the issue:

- /plans/:siteId ✅
- /plans/my-plan/:siteId ✅
- /settings/general/:siteId ✅
- /help 🚫 (maybe this is expected since it's not clear what site you're on; can you confirm @fditrapani or @michaeldcain?)
- /home/:siteId ✅
- /me/purchases/:siteId/:receiptId 🚫 
- /me/purchases 🚫 (@fditrapani added this in the issue, but since that view includes all sites, it seems like this is actually fine?)
- /domains/manage/:siteId ✅ 
- /domains/manage ✅ 

I'll leave it to @fditrapani and @michaeldcain to decide if those three remaining routes need this feature (or possibly if they should be their own PRs).

michaeldcain commented:
Thanks for sticking with the revisions @Aurorum!

> /help 🚫 (maybe this is expected since it's not clear what site you're on)

Can you add it to the bottom of the ["Most Helpful Articles"](https://github.com/Automattic/wp-calypso/blob/e4649a2418d1f878aea2d23b40e76875cae01ea1/client/me/help/main.jsx#L46)? I don't think we need to check for a purchase in this case, since it's a general support page and people might have questions about our policies pre-purchase.

> /me/purchases
> /me/purchases/:siteId/:receiptId

It's already [in the Purchases section](https://github.com/Automattic/wp-calypso/blob/81695512c30add75b1cb3cdde95f84d0f8677f1f/client/blocks/inline-help/contextual-help.js#L254), but can you update the title to the full "Managing Purchases, Renewals, and Cancellations"?

Aurorum commented:
Just wanted to say that I'll update this tomorrow - I've been taking a short break for a bit, thanks for the patience. :)

michaeldcain commented:
@Aurorum: no problem at all - you're the one helping us!

Aurorum committed:
Update title for Purchases screen

Aurorum committed:
Add to Help page

Aurorum commented:
I've made those two changes now. :) 

<img width="815" alt="Screenshot 2020-09-26 at 12 37 39" src="https://user-images.githubusercontent.com/43215253/94339854-143f8900-fff5-11ea-97a4-c97b5db9c45f.png">

michaeldcain commented on client/blocks/inline-help/inline-help-search-results.jsx:
```diff
@@ -123,13 +130,26 @@ function HelpSearchResults( {
 
+		// Unless searching with Inline Help or on the Purchases section, hide the
+		// "Managing Purchases" documentation link for users who have not made a purchase.
+		if ( post_id === 111349 && ! isSearching && ! hasPurchases && sectionName !== 'purchases' ) {
```
`! isSearching` only applies while the search is happening (not once the results are shown). If a user without a purchase searches "purchases", "refunds", etc… then this document will be hidden. `hasAPIResults` should fix that.
```suggestion
		if ( post_id === 111349 && ! isSearching && ! hasAPIResults && ! hasPurchases && sectionName !== 'purchases' ) {
```

Aurorum replied:
Thank you - that makes sense! :)

michaeldcain reviewed (approved):
This is looking good. I suggested one minor fix related to searching, but if you're okay with the change, I think we can merge this.

Thanks again for all the help on this.

michaeldcain committed:
Show link if result of a search query

michaeldcain committed:
Linting

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/4610342

Thank you @Aurorum for including a screenshot in the description! This is really helpful for our translators.

getdave commented:
Just a minor FYI @michaeldcain that there are a set of e2e tests which cover the Inline Help. 

* https://github.com/Automattic/wp-calypso/blob/5f26932f28e8e20930e59f079c4994e2c7dc6314/test/e2e/specs/wp-my-home-support-search-spec.js
* https://github.com/Automattic/wp-calypso/blob/5f26932f28e8e20930e59f079c4994e2c7dc6314/test/e2e/specs/wp-inline-help-support-search-spec.js

If your change is important to end-users then it might be worth updating the tests to account for this.

a8ci18n commented:
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/4610342).

jsnajdr commented on client/blocks/inline-help/inline-help-search-results.jsx:
```diff
@@ -123,13 +130,32 @@ function HelpSearchResults( {
+			! isSearching &&
+			! hasAPIResults &&
+			! hasPurchases &&
+			sectionName !== 'purchases'
```
Hi @Aurorum :wave: We're currently refactoring this component to use React Query (#55753) and this condition caught my attention.

Why is there the `! isSearching && ! hasAPIResults` condition? Do we want to hide the "Manage Purchase" link only when search is not in progress and when there are no results?

It would make more sense to me if the condition depended only on the `hasPurchases` and `sectionName` values.

Aurorum replied:
Hi! I can't remember for certain, but I'm pretty sure that the intention was that people who were actively searching for things to do with purchasing should still be able to see the "Managing Purchases" documentation even if they haven't made a purchase.

jsnajdr replied:
I think I understand the intent now: we want to hide the "Manage Purchases" article when it's part of the contextual results, like when in the `plans` section, but don't want to hide when it's a part of actual search results from the server. When I'm searching for "manage purchases", this article is one of the results returned by the server. We want to always show that.

Is that right?

Aurorum replied:
Yes, I believe that's correct.

jsnajdr replied:
In that case, it could be implemented by filtering only the "contextual results" array, probably in the related selector, and leaving the "search results" array intact. @flootr that should be easy to implement now when they are in fact separate arrays.
